### ISSUE TITLE
Add SLA Trigger conditions allowing to take action if a given Execution doesn't satisfy SLA criteria (max. allowed duration, nr of executions by date)

### ISSUE THREAD

**Issue by anna-geller** [kestra-io/kestra#1246]
### Feature description

TL;DR timeouts already solve the problem on a task level. This issue is a proposal to add one or more additional Trigger conditions to allow taking action (such as terminating a Flow's Execution and sending an alert about missed SLA) when the desired SLA duration is exceeded

## Problem 

Users can currently set **timeouts** on a task level to ensure a maximum allowed runtime. This concept is not available for **Flows**, and the best way to tackle it is by using **Triggers with an SLA condition** based on a given SLA condition.

### Duration SLA condition

```yaml
id: runtimeSLA
namespace: prod.monitoring
description: Zombie executions that get stuck

tasks:
  - id: sendAlertOnZombieTask
    type: io.kestra.plugin.notifications.slack.SlackExecution
    url: "{{envs.slack_webhook}}"
    channel: "#general"
    executionId: "{{trigger.executionId}}"

  - id: failExecution
    type: io.kestra.core.tasks.executions.Fail
    executionId: "{{trigger.executionId}}" # this will not work atm, but an illustrative example

triggers:
  - id: listen
    type: io.kestra.core.models.triggers.types.Flow
    conditions:
      - type: io.kestra.core.models.conditions.types.SLA
        duration: PT24H
      - type: io.kestra.core.models.conditions.types.ExecutionNamespaceCondition
        namespace: prod
        prefix: true
```

### Time-based aka `CronSLA` condition

SLA to define that there should be e.g. at least 1 successful Execution of a given flow ID every day by 9 AM. 

Pseudo code:

```yaml
triggers:
  - id: listen
    type: io.kestra.core.models.triggers.types.Flow
    conditions:
      - type: io.kestra.core.models.conditions.types.ExecutionStatusCondition
        in:
          - SUCCESS
      - type: io.kestra.core.models.conditions.types.ExecutionFlowCondition
        namespace: prod
        flowId: importantFlow
      - type: io.kestra.core.models.conditions.types.CronSLA
        executionsCount: 1
        cron: 0 9 * * *
```

### Freshness condition

Some community members would like to see freshness SLA, similar to how dbt handles that: https://docs.getdbt.com/docs/build/sources#snapshotting-source-data-freshness 

While technically, this is already possible to implement today using Query triggers, it might be worth considering either:
1) Making it easier -- some abstraction for how "freshness" of Execution or a given Metric (cc @Ben8t ) should be defined
2) Documenting how freshness policies can be implemented using Query triggers e.g.:
 
```yaml
id: freshness
namespace: dev

tasks:
  - id: action-if-data-not-fresh
    message: "{{ trigger.rows }}" # show the latest updated timestamp as opposed to what was expected

triggers:
  - id: watch
    type: io.kestra.plugin.jdbc.snowflake.Trigger
    interval: "PT5M"
    sql: |
        select max(_etl_loaded_at) as max_loaded_at, convert_timezone('UTC', current_timestamp()) as snapshotted_at
        from raw.jaffle_shop.orders
```

**Comment by Skraye** [comment]
Note that : 
```  - id: sla
    type: io.kestra.core.tasks.flows.Sequential
    timeout: PT10S
    tasks:
      - id: task-sleep
        type: io.kestra.core.tasks.scripts.Bash
        commands:
          - sleep 30
```
Should be doing the job, but isn't working.

**Comment by tchiotludo** [comment]
yes timeout is not working on flowable :cry:

**Comment by Ben8t** [comment]
Also adding SLA on metrics

**Comment by Ben8t** [comment]
Some food for thoughts :

- being able to look at tasks metric (if tasks.metric > X then fail/warning/retry/alert)
- being able to look up for previous executions and compute agg -> if task.metric < variance(all_tasks.metric) then Y

**Comment by Ben8t** [comment]
linked to #1373

**Comment by anna-geller** [comment]
based on a suggestion from @Melkaz it's worth considering implementing flow-level timeouts (https://github.com/kestra-io/kestra/issues/1373) as an SLA trigger that can perform various actions:

```yaml
id: runtimeSLA
namespace: prod.monitoring
description: Zombie executions that get stuck

tasks:
  - id: sendAlertOnZombieTask
    type: io.kestra.plugin.notifications.slack.SlackExecution
    url: "{{envs.slack_webhook}}"
    channel: "#general"
    executionId: "{{trigger.executionId}}"

  - id: failExecution
    type: io.kestra.core.tasks.executions.Fail
    executionId: "{{trigger.executionId}}" # this will not work atm, but an illustrative example

triggers:
  - id: listen
    type: io.kestra.core.models.triggers.types.Flow
    conditions:
      - type: io.kestra.core.models.conditions.types.SLA
        duration: PT24H
      - type: io.kestra.core.models.conditions.types.ExecutionNamespaceCondition
        namespace: prod
        prefix: true
```

**Comment by ale13jo** [comment]
Hi team, it would be great to have:

- Hold the execution outputs for re run, since the task can vary from non-return point, to avoid being purged perhaps new STATUS can be handled into a separate flow, like INCOMPLETE or something.

**Comment by anna-geller** [comment]
it will be addressed with a system flow https://github.com/kestra-io/kestra/issues/3350
